src/observable/fromArrayLike.ts

```diff
@@ -0,0 +1,69 @@
+import {Scheduler} from '../Scheduler';
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+
+export class ArrayLikeObservable<T> extends Observable<T> {
+
+  private _mapFn: (x: number, y: any) => T;
+
+  static create<T>(arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, scheduler?: Scheduler) {
+    return new ArrayLikeObservable(arrayLike, mapFn, thisArg, scheduler);
+  }
+
+  static dispatch(state: any) {
+
+    const { arrayLike, index, count, mapFn, subscriber } = state;
+
+    if (index >= count) {
+      subscriber.complete();
+      return;
+
+    }
+
+    let result = mapFn ? mapFn(index, arrayLike[index]) : arrayLike[index];
+    subscriber.next(result);
+
+    if (subscriber.isUnsubscribed) {
+      return;
+    }
+
+    state.index = index + 1;
+
+    (<any> this).schedule(state);
+  }
+
+  // value used if Array has one value and _isScalar
+  value: any;
+
+  constructor(public arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, public scheduler?: Scheduler) {
+    super();
+    if (!scheduler && arrayLike.length === 1) {
+      this._isScalar = true;
+      this.value = arrayLike[0];
+    }
+    if (mapFn) {
+      this._mapFn = mapFn.bind(thisArg);
+    }
+  }
+
+  _subscribe(subscriber: Subscriber<T>): Subscription | Function | void {
+    let index = 0;
+    const arrayLike = this.arrayLike;
+    const count = arrayLike.length;
+    const mapFn = this._mapFn;
+    const scheduler = this.scheduler;
```

Comment:
I liked way to shorten this kind of assignment in our codebase like
`const { arrayLike, count, mapFn... } = this`
can those be applicable here too?